### ISSUE TITLE
has_rocm_gpu() fails

### ISSUE THREAD

**Issue by michel2323** [JuliaGPU/AMDGPU.jl#564]
`has_rocm_gpu()` fails with
```
julia> AMDGPU.has_rocm_gpu()
ERROR: ArgumentError: Unknown component :hsa
Stacktrace:
 [1] functional(component::Symbol)
   @ AMDGPU /disk/mschanen/julia_depot/packages/AMDGPU/nMzRq/src/utils.jl:133
 [2] has_rocm_gpu()
   @ AMDGPU /disk/mschanen/julia_depot/packages/AMDGPU/nMzRq/src/utils.jl:138
 [3] top-level scope
   @ REPL[6]:1
```

on systems without AMD GPUs. `AMDGPU.functional()` works as expected. I am not sure if this is by design. In CUDA.jl `has_cuda_gpu()` works on systems without CUDA.

**Comment by michel2323** [comment]
It always fails in `v0.8.x`. Either readd `:hsa` to `functional(component::Symbol)` or remove calls to `functional(:hsa)` seems to be the solution.

**Comment by pxl-th** [comment]
Will be fixed by #565
